src/streamsync/core.py

```diff
@@ -928,7 +941,7 @@ def _check_proposed_session_id(self, proposed_session_id: Optional[str]) -> bool
             return True
         return False
 
-    def get_new_session(self, cookies: Optional[Dict] = None, headers: Optional[Dict] = None, proposed_session_id: Optional[str] = None) -> Optional[StreamsyncSession]:
+    def get_new_session(self, cookies: Optional[Dict] = None, headers: Optional[Dict] = None, proposed_session_id: Optional[str] = None, components: Optional[Dict[str, Any]] = None) -> Optional[StreamsyncSession]:
```

Comment:
We won't have to give the list of component instanciating the session.

Reply:
I agree that this shouldn't happen, should we mirrored in the same way that we handle initial state.